src/main/java/edu/cornell/kfs/coa/document/CUAccountMaintainableImpl.java

```diff
@@ -68,11 +81,27 @@ public void saveBusinessObject() {
         if (isClosingAccount) {
             SpringContext.getBean(AccountReversionTrickleDownInactivationService.class).trickleDownInactivateAccountReversions((Account) getBusinessObject(), getDocumentNumber());
         }
+        
+		// KFSPTS-3877 save xml again so that object id is persisted, this way
+		// notes on the doc will be retrieved form the DB instead of xml doc
+		// content and will display any future notes on account edits as well
+        try {
+            Document document = SpringContext.getBean(DocumentService.class).getByDocumentHeaderId(getDocumentNumber());
+            Account xmlaccount = (Account) ((MaintenanceDocument) document).getNewMaintainableObject().getBusinessObject();
+            if (ObjectUtils.isNull(xmlaccount.getObjectId()) &&( KFSConstants.MAINTENANCE_NEW_ACTION.equals(getMaintenanceAction()) || KFSConstants.MAINTENANCE_COPY_ACTION.equals(getMaintenanceAction()))) {
+                ((MaintenanceDocument) document).getNewMaintainableObject().setBusinessObject(account);
+                SpringContext.getBean(DocumentService.class).saveDocument(document);
```

Comment:
I have tested in my local environment and I did not run into any exception. I have followed an example from VendorMaintainableImpl (line 166) with this one. The problem is that the objectId is not persisted in the XML and that is why the notes don't display from the database but rather from the XML doc content. That is causing any notes added on account edit not to be displayed on the doc that originally created the account. Please let me know if you can think of a better way to do this and I am happy to change it. Thank you very much!